### PR TITLE
Fix coverity issues for unused values

### PR DESCRIPTION
**Pull request by siteshwar**
Resolves: CID#305912, CID#253749, CID#253822, CID#253739, CID#253552

### CODE REVIEW

File: src/cmd/ksh93/edit/edit.c [hunk 1]
```diff
@@ -1309,7 +1309,7 @@ int ed_histgen(Edit_t *ep, const char *pattern) {
     if (ac > 0) {
         l = ac;
         argv = av = (char **)stkalloc(ep->sh->stk, (ac + 1) * sizeof(char *));
-        for (mplast = 0; l >= 0 && (*av = (char *)mp); mplast = mp, mp = mp->next, av++) {
+        for (; l >= 0 && (*av = (char *)mp); mp = mp->next, av++) {
```
Comment (krader1961) [inline src/cmd/ksh93/edit/edit.c]: Isn't the `mplast = mp;` at line 1300 also pointless?

File: src/lib/libcmd/wclib.c [hunk 2]
```diff
@@ -390,7 +390,6 @@ int wc_count(Wc_t *wp, Sfio_t *fd, const char *file) {
                         }
                         while (mbc(c) && ((c & WC_ERR) || (c & 7) == 0)) c = type[*cp++];
                         if (eol(c) && (cp > endbuff)) {
-                            c = WC_MB | WC_ERR;
```
Comment (krader1961) [inline src/lib/libcmd/wclib.c]: Note that I recently fixed a related bug on line 391 which had an always true condition: `((c | WC_ERR) || (c & 7) == 0)`. I now have very low confidence this code does the right thing.

**Comment by krader1961** [comment]
LGTM

File: src/cmd/ksh93/edit/edit.c (hunk 1 above)
Comment (siteshwar) [reply src/cmd/ksh93/edit/edit.c]: Probably not. It's value can be used in the next iteration of enclosing `for` loop.

**Pull request merged**